src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -312,6 +323,60 @@ class EthService(
     }
   }
 
+  def getBalance(req: GetBalanceRequest): ServiceResponse[GetBalanceResponse] = {
+    Future.successful {
+      withAccount(Address(req.address), req.block) { account =>
+        GetBalanceResponse(account.balance)
+      }
+    }
+  }
+
+  def getStorageAt(req: GetStorageAtRequest): ServiceResponse[GetStorageAtResponse] = {
+    import io.iohk.ethereum.rlp.UInt256RLPImplicits._
+
+    val byteArrayUInt256Serializer = new ByteArrayEncoder[UInt256] {
+      override def toBytes(input: UInt256): Array[Byte] = input.bytes.toArray[Byte]
+    }
+
+    val rlpUInt256Serializer = new ByteArraySerializable[UInt256] {
+      override def fromBytes(bytes: Array[Byte]): UInt256 = ByteString(bytes).toUInt256
+      override def toBytes(input: UInt256): Array[Byte] = input.toBytes
+    }
+
+    Future.successful {
+      withAccount(Address(req.address), req.block) { account =>
+        val storageMpt =
+          MerklePatriciaTrie[UInt256, UInt256](account.storageRoot.toArray[Byte],
+            blockchainStorages.nodeStorage, crypto.kec256(_: Array[Byte]))(HashByteArraySerializable(byteArrayUInt256Serializer), rlpUInt256Serializer)
+        GetStorageAtResponse(storageMpt.get(UInt256(req.position)).getOrElse(UInt256(0)).bytes)
+      }
+    }
+  }
+
+  def getTransactionCount(req: GetTransactionCountRequest): ServiceResponse[GetTransactionCountResponse] = {
+    Future.successful {
+      withAccount(Address(req.address), req.block) { account =>
+        GetTransactionCountResponse(account.nonce)
+      }
+    }
+  }
+
+  private def withAccount[T](address: Address, blockParam: BlockParam)(f: Account => T): Either[JsonRpcError, T] = {
+    withStateMpt(blockParam) { stateMpt =>
+      val account = stateMpt.get(kec256(address.toArray)).getOrElse(Account.Empty)
```

Comment:
So I added the serializers to `Address` comp object because of how it was done for `Account`. But now that I think of it, those serializers should also reside in `mpt` package object. It makes more sense to me for the storage to know about the objects it stores, rather than the objects knowing how they are stored.

[This remark](https://github.com/input-output-hk/etc-client/pull/200#discussion_r119828406) still holds, though.